packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkCodecUtilities.ts

```diff
@@ -176,8 +177,14 @@ export interface ChunkDecoder {
 	/**
 	 * Read from stream, updating the offset.
 	 *
+	 * For fields that support incremental encoding, the encoded data can be retrieved via `getIncrementalFieldBatch`.
+	 *
 	 * @returns a TreeChunk made from the data from `stream`.
 	 * This chunk is allowed to reference/take ownership of content it reads from the stream.
 	 */
-	decode(decoders: readonly ChunkDecoder[], stream: StreamCursor): TreeChunk;
+	decode(
+		decoders: readonly ChunkDecoder[],
+		stream: StreamCursor,
+		getIncrementalFieldBatch?: (fieldKey: string) => EncodedFieldBatch,
```

Comment:
the type `(fieldKey: string) => EncodedFieldBatch` is not very self explanatory to me. I don't get what its for. Factoring out this into its own type so you can put documentation on it might help.

Another option would be to document this method parameter to say what its for as part of this methods' docs. You did add a comment, but it doesn't make a lot of sense to me: I don't see why a field key would be provided. Maybe that parameter is supposed to be a summaryRefId instead?

Reply:
Added it to a separate interface called `IncrementalDecoder` and added comments.